src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
 
 using namespace SmartRedis;
 
+
+// Return a pointer to a new Client.
+// The caller is responsible for deleting the client via DeleteClient().
 extern "C"
-void* SmartRedisCClient(bool cluster)
+void *SmartRedisCClient(bool cluster)
 {
-  /* Return a pointer to a new Client.
-  The user is responsible for deleting the client.
-  */
-  Client* s = new Client(cluster);
-  return (void*)s;
+  Client *s = NULL;
+
+  try {
+    s = new Client(cluster);
+  } catch (...) {
```

Comment:
If we are going to return ``NULL`` on all exceptions, we should probably split this up because you can have memory leak if ``new`` operator succeeds but some redis error is thrown.  I would split into a ``bad_alloc`` catch and then have a ``...`` catch which would have a delete operation to prevent memory leaks.

Reply:
Good point! I will split this.